### PR TITLE
Run tests with xvfb-run on Linux

### PR DESCRIPTION
**Pull request by ChrisThrasher**
Fixes build errors in https://github.com/SFML/SFML/pull/2302.

### CODE REVIEW

File: buildfactories.py [hunk 1]
```diff
@@ -135,9 +135,13 @@ def get_build_step(link, type, options = []):
     from buildbot.process.properties import Interpolate
     from buildbot.steps.shell import Compile
 
+    prefix = ''
     suffix = ''
     target = 'install'
 
+    if 'linux' in options:
+        prefix = 'xvfb-run -a '
```
Comment (eXpl0it3r) [inline buildfactories.py:143]: FreeBSD unfortunately doesn't have `xvfb-run`

**Comment by ChrisThrasher** [comment]
In https://github.com/SFML/SFML/pull/2302 I added a new option called `SFML_ENABLE_OPENGL_TESTS` which enables tests that require a display. We could just set this option to `OFF` for all extended CI pipeline jobs. This is overkill but honestly not a big deal since we get great coverage running these tests on all possible jobs in the non-extended CI pipeline. That would let us wrap up this issue fairly quickly and move onto actually writing the tests we want to write.

**Comment by eXpl0it3r** [comment]
I've already gotten xvfb installed on Debian, so builds on that and Windows should be fine.  
I think for the Raspberry Pi I didn't manage to get it working either, so I wouldn't mind not running it there.

**Comment by binary1248** [comment]
Display tests on Linux (Debian) are enabled by d0e9c35aac87a6d40c1498fd13ce4be171a19068. We still shouldn't attempt to run the display tests on anything else than that (Android, FreeBSD, etc.). The only safe workers to run display tests on are Windows, macOS and Debian right now.

**Comment by binary1248** [comment]
The whitelist [here](https://github.com/SFML/SFML/blob/474de4d16e570b9cb4258d85d5f9142a26e22202/test/CMakeLists.txt#L53) should be adjusted accordingly to enable by default only on Windows, macOS and Linux.

**Pull request closed without merging**